File: Code/new_fitervari_backend/src/main/java/at/htl/control/ExerciseRepository.java
package at.htl.control;

import at.htl.model.Exercise;
import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class ExerciseRepository implements PanacheRepository<Exercise> {
    public Exercise save(Exercise entity) {
        return getEntityManager().merge(entity);
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/repository/ExerciseRepository.kt
package at.htl_leonding.repository

import at.htl_leonding.model.Exercise
import io.quarkus.hibernate.orm.panache.PanacheRepository
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject
import javax.persistence.EntityManager

@ApplicationScoped
class ExerciseRepository : PanacheRepository<Exercise>{
    @Inject
    lateinit var em: EntityManager

    fun save(exercise: Exercise): Exercise {
        return em.merge(exercise)
    }
}